hw/top_earlgrey/data/ip/chip_entropy_src_testplan.hjson

```diff
@@ -0,0 +1,195 @@
+            - Configure realistic health test threshold values for FIPS/CC compliant mode.
+            - Enable ENTROPY_SRC in FIPS mode.
+            - Enable CSRNG.
+            - Enable both EDNs in auto request mode.
+            - Trigger the execution of various cryptographic hardware blocks consuming entropy to stress test the entropy complex.
+            - Verify that all entropy consuming endpoints can finish their operations and do not hang.
+            - Repeat the procedure with different health test window sizes for FIPS mode.
+            - Disable the entropy complex.
+            - Configure very stringent health test threshold values for FIPS mode.
+            - Configure a low alert threshold value in the ALERT_THRESHOLD register.
+            - Enable ENTROPY_SRC in FIPS mode.
+            - Enable CSRNG.
+            - Enable both EDNs in auto request mode and with low values for the generate length parameter to request fresh entropy more often.
+            - Trigger the execution of various cryptographic hardware blocks consuming entropy to stress test the entropy complex.
+            - Verify that ENTROPY_SRC triggers a recoverable alert and sets the RECOV_ALERT_STS.ES_MAIN_SM_ALERT bit.
+            - Verify that various entropy consuming endpoints hang as ENTROPY_SRC stops generating entropy after triggering the recoverable alert.
+            '''
+      features: [
+        "ENTROPY_SRC.MODE.FIPS",
+        "ENTROPY_SRC.HEALTH_TESTS",
+      ]
+      stage: V3
+      si_stage: SV3
+      tests: []
+      bazel: []
+    },
+    {
+      name: chip_sw_entropy_src_validation
+      desc: '''
+            Enable validation of the four channels of the digital noise source / physical true random number generator (PTRNG) at the ENTROPY_SRC input, and verify that ENTROPY_SRC allows observing raw entropy through the observe FIFO while serving entropy to consumers.
+
+            Procedure:
+            - Disable the entropy complex.
+            - Configure realistic health test threshold values for FIPS/CC compliant mode.
+            - Enable ENTROPY_SRC in FIPS mode.
+            - Enable Firmware Override - Observe.
+            - Enable CSRNG.
+            - Enable both EDNs in auto request mode.
+            - Trigger the execution of various cryptographic hardware blocks consuming entropy to stress test the entropy complex.
+            - Verify that all entropy consuming endpoints can finish their operations and do not hang.
+            - Verify that the observe FIFO depth is non-zero and read out post-health test entropy for validation testing.
+            - Repeat the procedure four times with the ENTROPY_SRC.RNG_BIT_ENABLE feature being enabled (once for every channel).
+
+            Notes for silicon targets:
+            - This test requires access to the ENTROPY_SRC.FW_OV.OBSERVE feature, which is gated by OTP.
+              This test might thus need to be done in a secure facility.
```

Comment:
Like other tests, we might also be able to run this in PROD eventually. Depending on when we're going to activate the corresponding OTP fuses exactly.